builder/dispatchers.go

```diff
@@ -552,3 +605,59 @@ func volume(b *builder, args []string, attributes map[string]bool, original stri
 	}
 	return nil
 }
+
+// ARG --default=bar --description='my variable' foo
+//
+// Adds the variable foo to the trusted list of variables that can be passed
+// to builder using the --build-arg flag for expansion/subsitution or passing to 'run'.
+// Dockerfile author may optionally set a default value and description of this variable.
+func arg(b *builder, args []string, attributes map[string]bool, original string) error {
+	const (
+		def  = "default"
+		desc = "description"
+	)
+
+	if len(args) != 1 {
+		return fmt.Errorf("ARG requires exactly one argument")
+	}
+
+	// 'arg' accepts the default and description flags
+	if flag := b.BuilderFlags.AddString(def, ""); flag == nil {
```

Comment:
you don't need the "if" - just call AddString() and if there's an error then the Parse() down below will catch it.  This saves you from having to add an "if" around each AddXXX() call.